### PR TITLE
Dev-dashboard-pages

### PR DESCRIPTION
**Pull request by YuejiaoShi**
preview

### CODE REVIEW

File: frontend/app/_components/AuthComponents/LogIn.js [hunk 1]
```diff
@@ -23,17 +22,18 @@ const LogIn = () => {
...
+        Cookies.set("userId", userInfo.id);
+        toast.success("Logged in successfully! 🎉");
+
         const dashboardPath = getUserPathByRole(userRole);
         window.location.replace(dashboardPath);
       });
-    } catch (error) {
-      setFormError("Login failed. Please try again.");
-    }
+
```
Comment (AlejandroGispert) [inline frontend/app/_components/AuthComponents/LogIn.js:36]: why did you remove the try and catch, isnt that good for detecting errors there?

**Review by AlejandroGispert — APPROVED** [review]

File: frontend/app/_components/AuthComponents/LogIn.js (hunk 1 above)
Comment (YuejiaoShi) [reply frontend/app/_components/AuthComponents/LogIn.js:36]: I've handle these responds in the functions at resHandler.js. So I donot need the try catch here.

File: frontend/app/dev-dashboard/projects/[id]/page.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+              <strong className='mr-2 text-base'>Start Date:</strong> {project.startDate}
+            </p>
+            <p className='flex items-center mb-4'>
+              <CalendarTodayOutlinedIcon className='mr-3 text-primary-blue-dark text-xl md:text-2xl' />
+              <strong className='mr-2 text-base'>Deadline:</strong> {project.deadline}
+            </p>
+          </div>
+        </div>
+
+        {/* {project.demoUrl && ( */}
```
Comment (magdazelena) [inline frontend/app/dev-dashboard/projects/[id]/page.js:117]: 👀

File: frontend/tailwind.config.js [hunk 3]
```diff
@@ -91,8 +104,15 @@ module.exports = {
...
+			sm: '640px',
+			md: '768px',
+			lg: '1024px',
+		},
+	  },
+	},
+	plugins: [],
 };
+
+//   plugins: [require("tailwindcss-animate")],
```
Comment (magdazelena) [inline frontend/tailwind.config.js:118]: 👀

**Review by magdazelena — APPROVED** [review]
whoa, excellent work! added some eyes for the commented code (maybe it's there for a reason, idk ;)). This is really impressive and I see a lot of good development patterns, good job!

One thing I would suggest though is splitting PRs into smaller chunks. It is quite tedious to review this much code and easy to miss parts and bits. I understand this is a one feature and if you don't want to make separate PRs for breaking code or stale components (before implementing) it's weird to split. I would, however suggest stacking flow: https://www.stacking.dev/ for example - so small prs based on one another. This way you can get smaller chunks without risk of getting bad code into the main!

File: frontend/app/dev-dashboard/projects/[id]/page.js (hunk 2 above)
Comment (YuejiaoShi) [reply frontend/app/dev-dashboard/projects/[id]/page.js:117]: That's because we currently donot have demoUrl at database, so I just  use a fake image here to simulate the demo area.😊

File: frontend/tailwind.config.js (hunk 3 above)
Comment (YuejiaoShi) [reply frontend/tailwind.config.js:118]: This line is from another teammate, but it cause error and he didn't need it anymore. But I'd like to leave it here for now in case he might want it back later.
